### PR TITLE
Sanitizer

### PR DESCRIPTION
**Pull request by kalilistic**
Sanitizer from discussion in discord. Intended for cleaning up strings to use with imgui to avoid ???. Accepts individual or lists of strings and language overrides.

### CODE REVIEW

File: Dalamud/Plugin/Sanitizer/Sanitizer.cs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+using System.Text;
+
+namespace Dalamud.Plugin.Sanitizer
+{
+    /// <summary>
+    /// Sanitize strings to remove soft hyphens and other special characters.
+    /// </summary>
+    public class Sanitizer : ISanitizer
+    {
+        private readonly KeyValuePair<string, string> softHyphen = new KeyValuePair<string, string>("\u0002\u0016\u0001\u0003", string.Empty);
```
Comment (NotAdam) [inline Dalamud/Plugin/Sanitizer/Sanitizer.cs:13]: why? just put it in a dict by language

**Comment by NotAdam** [comment]
also whats the point of tests that don't test anything that it actually does

**Comment by kalilistic** [comment]
Will do a new PR later with changes.

**Comment by kalilistic** [comment]
For ackwell.

**Comment by kalilistic** [comment]
- Replaced most hard-coded characters with 0x20 filter
- Replaced kvp lists with dictionaries
- Fixed unit test strings so can see characters in github
- Updated unit tests to use theory/inline data

**Review by goaaats — APPROVED** [review]

**Pull request merged**